### PR TITLE
⬆️ Combined github actions updates

### PR DESCRIPTION
**Pull request by s-weigand**
Combined update of github actions

### Change summary

- [Bump actions/cache from 3 to 4](https://github.com/glotaran/pyglotaran/commit/b439d232a82c641ab9d2505e2d05c4c6118bc13e)
- [Bump codecov/codecov-action from 3 to 4](https://github.com/glotaran/pyglotaran/commit/01e67573a86736b0fbdf412504f2ba52115e0ffc)
- [Bump github/codeql-action from 2 to 3](https://github.com/glotaran/pyglotaran/commit/84aede05cec1f42e7f3a83abe0046ac5f55c2716)
- [Bump actions/upload-artifact from 3 to 4](https://github.com/glotaran/pyglotaran/commit/5922f81350ea55a4c5d039465ae2de93d1b94d96)
- [Bump actions/download-artifact from 3 to 4](https://github.com/glotaran/pyglotaran/commit/c0529a78a70261e179c86de83506596ba98346d3)
- [🩹 Adapt to changes in artifact actions](https://github.com/glotaran/pyglotaran/commit/03b0c56e8d359236c00900524a72f3bd7b7eb247)

### Checklist

- [x] ✔️ Passing the tests (mandatory for all PR's)
- [x] 👌 Closes issue (mandatory for ✨ feature and 🩹 bug fix PR's)

### Closes issues

closes #1409 
closes #1410
closes #1411
closes #1417
closes #1426

### CODE REVIEW

File: .github/workflows/integration-tests.yml [hunk 1]
```diff
@@ -51,17 +51,43 @@ jobs:
...
+
+      - name: Upload Example Plots Artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: example-plots
+          path: example-plots
+          overwrite: true
+
+      - name: Delete Intermediate artifacts
+        #  TODO: Switch back to GeekyEggo/delete-artifact after
```
Comment (jsnel) [inline .github/workflows/integration-tests.yml:85]: Create an issue for that so you are reminded in case it takes a while?

**Review by jsnel — APPROVED** [review]
Looks good.

Splitting up collecting plot artifacts 👍

File: .github/workflows/integration-tests.yml (hunk 1 above)
Comment (s-weigand) [reply .github/workflows/integration-tests.yml:85]: I created issue #1439

**Review by jsnel — APPROVED** [review]
Lgtm

**Pull request merged**